### PR TITLE
fix: restore the arithmetic safety net this daemon claimed to have

### PR DESCRIPTION
**Pull request by kirushik**
Two independent causes left a money-handling daemon with neither the compile-time arithmetic lint nor the runtime overflow check its own configuration and documentation advertised.

1. `[profile.dev]` / `[profile.release]` lived in `daemon/Cargo.toml`, a non-root workspace member. Cargo silently ignores profile tables outside the workspace root, so `overflow-checks = true` (and `lto`, `codegen-units`, `panic = "abort"`) never applied. Verified against the real rustc invocation from `cargo build --release -vv`, not the manifest. Moved to the root `Cargo.toml`.

2. `clippy::arithmetic_side_effects` had been removed from the root `Cargo.toml` in ed941bc ("style: fix clippy and rustfmt warnings"), while ~12 `#[expect(clippy::arithmetic_side_effects)]` attributes stayed in the tree as inert markers. Re-enabled workspace-wide and triaged all 32 resulting violations.

Money and gas paths fixed properly (checked/fallible, domain errors):

- etherscan_client/types.rs: `Decimal::new(self.value as i64, ...)` wrapped a u128 into an i64. Ten units of an 18-decimal token is 10^19 base units, so an ordinary payment was recorded as a NEGATIVE amount. Now a checked, fallible conversion behind `EtherscanClientError::UnrepresentableAmount`.
- chain_client/polygon.rs: `u256_to_decimal` and `decimal_to_u256` substituted zero on failure, recording a real payment (or payout) as nothing. Both are now `Option`-returning, with domain errors at all four call sites.
- chain_client/polygon.rs: `U256::to::<u128>()` panics above u128::MAX on values taken verbatim from unvalidated Pimlico bundler responses. Replaced with a checked `u256_to_u128` returning `TransactionError::BuildFailed`.
- chain_client/polygon.rs: `calculate_max_cost_in_token` did unchecked U256 `+`/`*` on bundler-supplied gas quotes. Now fully checked, and lifted to a free function so it can be tested without a live provider.
- chain_client/asset_hub.rs, chain/executor.rs, state/swaps.rs: base-unit scaling via `amount / Decimal::new(1, decimals)` panics for decimals > 28 and on Decimal overflow, and `chain/executor.rs` then `.unwrap()`ed the `to_u128`. All three are now checked and fallible.
- chain/transactions_recorder.rs: invoice total, refund amount and the under/overpayment tolerance arithmetic now use `checked_*` and abandon the DB transaction via `TransactionsRecorderError::AmountOverflow`.
- types/common.rs: `PaginationParams::offset()` computed `(page - 1) * per_page` on an unvalidated query parameter — a u32 overflow that used to wrap silently and, with overflow-checks now actually on, would panic the handler.
- auth/token.rs, api/dev.rs, state.rs: `DateTime + Duration` and `TimeDelta::seconds/minutes/milliseconds` panic on out-of-range values derived from token claims, request bodies and config. All now use `checked_add_signed` / `try_*` and fail closed.
- types/invoice.rs, webhook_sender.rs, chain/executor.rs: saturating arithmetic where saturation is the correct semantics.

Remaining 12 sites are `#[expect(..., reason = "...")]` with a specific justification: fixed-offset clock arithmetic and test fixtures.

Docs corrected to match reality: conventions.md previously claimed `pedantic`, `arithmetic_side_effects` and the `shadow_*` family were enabled per-crate; none were.

Adds 29 tests covering every changed behaviour, including boundary cases at u128::MAX / one-over-max and Decimal's representable range.

### CODE REVIEW

File: daemon/src/auth/token.rs [hunk 1]
```diff
@@ -189,31 +189,57 @@ fn get_string_claim(
...
     let now = Utc::now();
-    let tolerance =
-        chrono::Duration::seconds(i64::try_from(clock_tolerance_secs).unwrap_or(i64::MAX));
-    now > claims.exp + tolerance
+    let Some(tolerance) = i64::try_from(clock_tolerance_secs)
+        .ok()
+        .and_then(chrono::TimeDelta::try_seconds)
+    else {
+        return true
+    };
```
[the remark this answers is not in the document]
Comment (kirushik) [reply daemon/src/auth/token.rs:208]: Obviously wrong: `cargo check` passes

File: daemon/src/chain/transfer_tracker.rs [hunk 2]
```diff
@@ -193,24 +202,30 @@ impl<T: ChainConfig, C: BlockChainClient<T> + 'static, D: DaoInterface + 'static
...
-                ),
+                Err(error) => {
+                    tracing::error!(
+                        %invoice_id,
+                        ?transaction_id,
+                        error = ?error,
+                        "Incoming transfer was not recorded; durable replay is required for guaranteed recovery"
+                    );
+                    return Err(error)
+                },
```
[the remark this answers is not in the document]
Comment (kirushik) [reply daemon/src/chain/transfer_tracker.rs:213]: Obviously wrong: `cargo check` passes

File: daemon/src/configs/types.rs [hunk 3]
```diff
@@ -355,7 +356,33 @@ pub struct PaymentsConfig {
...
+            .and_then(|lifetime| Utc::now().checked_add_signed(lifetime))
+            .is_some();
+
+        if !valid_lifetime {
+            return Err(
+                PaymentsConfigError::InvalidInvoiceLifetime {
+                    value: self.invoice_lifetime_millis,
+                },
+            )
+        }
```
[the remark this answers is not in the document]
Comment (kirushik) [reply daemon/src/configs/types.rs:394]: Obviously wrong: `cargo check` passes
